### PR TITLE
Fix findall_record to return empty list for no matches

### LINKED ISSUE #10458: [BUG] `cudf::strings::findall_record` should return an empty list instead of null if no matches are found
**Describe the bug**
PR #9911 introduced an API `cudf::strings::findall_record`. This resolved #9856, which was a request for Spark's [regexp_extract_all](https://spark.apache.org/docs/latest/api/sql/index.html#regexp_extract_all) function.

Currently, `cudf::strings::findall_record` returns a null value if no matches are found. However, both Spark and Pandas return an empty list. I discovered this while working on #10226.

Spark SQL:
```
spark-sql> SELECT regexp_extract_all('Cat Monkey Monkey', 'Monkey', 0);
["Monkey","Monkey"]
Time taken: 0.051 seconds, Fetched 1 row(s)
spark-sql> SELECT regexp_extract_all('Cat', 'Monkey', 0);
[]
Time taken: 0.035 seconds, Fetched 1 row(s)
```

Pandas:
```python
>>> import pandas as pd
>>> ps = pd.Series(["Cat Monkey Monkey", "Cat"])
>>> print(ps.str.findall("Monkey"))
0    [Monkey, Monkey]
1                  []
dtype: object
```

**Expected behavior**
If no matches are found, the result should be an empty list instead of a null value.

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #10458 

Change the behavior of `cudf::strings::findall_record` to return an empty row when the search results in no matches for the corresponding input strings row. The previous behavior was returning a null row when no matches are found.

### CODE REVIEW

File: cpp/include/cudf/strings/findall.hpp [hunk 1]
```diff
@@ -72,11 +72,10 @@ std::unique_ptr<table> findall(
  *  [ ["b"]
  *    ["a","b","b"]
  *    ["a"]
- *    null ]
+ *    [] ]
  * @endcode
  *
- * A null output row results if the pattern is not found in the corresponding row
- * input string.
+ * A null output row occurs if the corresponding input row is null.
```
Comment (hyperbolic2346) [inline cpp/include/cudf/strings/findall.hpp:78]: Maybe add a small blurb about things that don't match resulting in an empty list. It is shown in the example, but might be worth calling out.

File: cpp/src/strings/search/findall_record.cu [hunk 2]
```diff
@@ -117,26 +117,6 @@ std::unique_ptr<column> findall_record(
...
-  if (valid_count == 0) {
-    return make_lists_column(0,
-                             make_empty_column(type_to_id<offset_type>()),
-                             make_empty_column(type_id::STRING),
-                             0,
-                             rmm::device_buffer{},
-                             stream,
-                             mr);
-  }
-
```
Comment (hyperbolic2346) [inline cpp/src/strings/search/findall_record.cu:139]: Stoptimizations are the best! 🎉

**Review by hyperbolic2346 — APPROVED** [review]
Nitpick at best. Happy to let it go either way.

**Review by bdice — APPROVED** [review]
Thank you @davidwendt! This looks good to me. I will continue work on #10459 once this is merged.

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**